samples/client-sdk/knocking/README.md

```diff
@@ -1 +1,94 @@
+
+```
+curl -H "Content-Type: application/json" \
+     -H "Authorization: Bearer $DAILY_API_KEY" \
+     -XPOST -d \
+     '{"name": "room-name",
+       "privacy": "private",
+       "properties" : {
+          "enable_knocking":true}}' \
+     https://api.daily.co/v1/rooms/
+
+```
+
+**Note:** This demo app assumes cameras will be turned on by default when a participant joins. This is the default setting for new rooms. The [`start_video_off`](https://docs.daily.co/reference/rest-api/rooms/config#start_video_off) setting should be `false`.
+
+3. An owner meeting token, which you can create with the following cURL command:
+
+```
+curl -H "Content-Type: application/json" \
+     -H "Authorization: Bearer $DAILY_API_KEY" \
+     -XPOST -d '{"properties":
+                  {"is_owner":true, "room_name":"room-name"}}' \
+     https://api.daily.co/v1/meeting-tokens
+```
+
+**Note:** You must set the `is_owner` configuration to `true`. If you do not, you will be able to join the call but will not have owner privileges to respond to guests knocking.
+
+**Warning:** Your API key and meeting token should be stored securely. Never hard-code them in your app or commit them to the version control history.
+
+## Run demo locally
+
+To run this demo locally, start by cloning this repo locally and navigate to the repo's directory. Next, you can either:
+
+- Open the `index.html` file in your browser, or;
+- Run the following commands to start a local server:
+
+```
+npm install
+npm run start
+```
+
+To test this as a owner and guest, have one window open for the owner and one window open for the guest.
+
+We recommend keeping your browser console open while testing to see any messages, errors, or events while they're happening.
+
+## How to use this demo
+
+### Join as a meeting owner
+
+Meeting owners can join private calls without knocking. When a guest knocks, owners can decide to let the guest into the call or reject their request.
+
+To join as a meeting owner, fill out the _Meeting owner form_ with your name, the Daily room URL, and your token.
+
+---
+
+![Owner view when guest is knocking to enter](./assets/owner.png)
+
+---
+
+Once you join, you will see your video and a waiting room list. Guests that have knocked to join your call will show in the list. If there are guests awaiting, you can press the "Allow access" or "Deny guest access" buttons to let them in or reject them.
+
+### Join as a guest
+
+Guests can knock to join a private call when the [`enable_knocking`](https://docs.daily.co/reference/rest-api/rooms/create-room#enable_knocking) room setting is enabled.
+
+To join as a guest, enter your name and the room URL in the _Guest form_. Do not join both forms in the same browser window. (Open a new tab for the Guest view.)
```

Comment:
I think we can hide the forms when the user is in a call to prevent this double-join-attempt from happening. It might also be nice to not show the Leave button until the user is _in_ the call. I've actually got an implementation of this that I did while testing locally (works great btw!) - mind if I file a PR against this branch with that update? It doesn't add any super new logic, just a couple of addition DOM hide/show methods.

Reply:
Sure thing, if you have it already, feel free to add!